universum/configuration_support.py

```diff
@@ -117,7 +132,38 @@ def __getitem__(self, item: str) -> Optional[str]:
         'bar'
         >>> cfg['test']
         """
-        return self.extras.get(item, self.__dict__.get(item, None))
+        return self._extras.get(item, self.__dict__.get(item, None))
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        """
+        This functions simulates dict-like legacy write access
+
+        :param key: client-defined key
+        :param value: client-defined value
+
+        >>> import warnings
+        >>> def set_cfg_and_get_warnings(c, k, v):
+        ...     with warnings.catch_warnings(record=True) as w:
+        ...         warnings.simplefilter("always")
+        ...         c[k] = v
+        ...         return w
+        >>> cfg = ProjectConfiguration(name='foo', my_var='bar')
+        >>> set_cfg_and_get_warnings(cfg, 'name', 'bar')  # doctest: +ELLIPSIS
+        [<warnings.WarningMessage object at ...>]
+        >>> set_cfg_and_get_warnings(cfg, 'directory', 'foo')  # doctest: +ELLIPSIS
+        [<warnings.WarningMessage object at ...>]
+        >>> set_cfg_and_get_warnings(cfg, '_extras', 'baz')
+        []
+        >>> set_cfg_and_get_warnings(cfg, 'test', 42)
+        []
+        >>> cfg
+        {'name': 'bar', 'directory': 'foo', 'my_var': 'bar', '_extras': 'baz', 'test': 42}
+        """
+        if key in self.__dict__ and key != '_extras':
+            warn("Re-defining the value of a read-only ProjectConfiguration field", UserWarning, 3)
```

Comment:
Well, the field is not read-only, it just has to be accessed using object-like access.
I would give an example of proper usage in the warning, like the following:
"Re-defining the value of ProjectConfiguration field. Please use var.command to set it instead of using var['command']."

Reply:
fixed